### PR TITLE
nautilus: rgw: Modify the error message of PutBucketAcls when the bucket does not exist

### PR DESCRIPTION
**Pull request by ybwang0211**
rgw:Modify the error message of PutBucketAcls when the bucket does not exist

When the bucket does not exist, the "NoSuchKey" error will be thrown after executing PutBucketAcls, but the more accurate error should be "NoSuchBucket", so we modified the "RGWRados::get_bucket_entrypoint_info" function and corrected the error code.

Signed-off-by: wangyingbin <wangyingbin@inspur.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/rgw/rgw_rados.cc [hunk 1]
```diff
@@ -8274,6 +8274,9 @@ int RGWRados::get_bucket_entrypoint_info(RGWSysObjectCtx& obj_ctx,
 			       bucket_entry, bl, objv_tracker, pmtime, pattrs,
 			       cache_info, refresh_version);
   if (ret < 0) {
+    if (ret == -2){
```
[the remark this answers is not in the document]
Comment (ybwang0211) [reply src/rgw/rgw_rados.cc:8277]: Do you mean to correct the error as follows:
  int ret = rgw_get_system_obj(this, obj_ctx, svc.zone->get_zone_params().domain_root,
			       bucket_entry, bl, objv_tracker, pmtime, pattrs,
			       cache_info, refresh_version);
  if (ret < 0) {
    return -ERR_NO_SUCH_BUCKET;
  }

[the remark this answers is not in the document]
Comment (theanalyst) [reply src/rgw/rgw_rados.cc:8277]: yep `if (ret == -ENOENT) return -ERR_NO_SUCH_BUCKET`

[the remark this answers is not in the document]
Comment (ybwang0211) [reply src/rgw/rgw_rados.cc]: I have revised it as required,Please check it.

**Comment by smithfarm** [comment]
No response; closing. Feel free to reopen once the change request has been addressed.

**Pull request closed without merging**